### PR TITLE
[Merged by Bors] - feat(Trigonometric): add `Complex.tan_eq_zero_iff'`

### PR DESCRIPTION
**Pull request by urkud**
This version assumes that the cosine is not zero.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by grunweg — APPROVED** [review]

File: Mathlib/Analysis/SpecialFunctions/Trigonometric/Complex.lean [hunk 1]
```diff
@@ -61,13 +61,15 @@ theorem sin_ne_zero_iff {θ : ℂ} : sin θ ≠ 0 ↔ ∀ k : ℤ, θ ≠ k * π
   rw [← not_exists, not_iff_not, sin_eq_zero_iff]
 #align complex.sin_ne_zero_iff Complex.sin_ne_zero_iff
 
+/-- Tangent of a complex number is equal to zero
+iff this number is equal to `k * π / 2` for an integer `k`.
+
+Note that this lemma takes into account that we use zero as the junk value for division by zero.
+See also `Complex.tan_eq_zero_iff'`.  -/
 theorem tan_eq_zero_iff {θ : ℂ} : tan θ = 0 ↔ ∃ k : ℤ, θ = k * π / 2 := by
```
Comment (sgouezel) [inline Mathlib/Analysis/SpecialFunctions/Trigonometric/Complex.lean:69]: Could you take this occasion to swap the right hand side to `k * π / 2 = θ`? Same thing in the next lemma.

File: Mathlib/Analysis/SpecialFunctions/Trigonometric/Complex.lean [hunk 2]
```diff
@@ -61,13 +61,15 @@ theorem sin_ne_zero_iff {θ : ℂ} : sin θ ≠ 0 ↔ ∀ k : ℤ, θ ≠ k * π
   rw [← not_exists, not_iff_not, sin_eq_zero_iff]
 #align complex.sin_ne_zero_iff Complex.sin_ne_zero_iff
 
+/-- Tangent of a complex number is equal to zero
```
Comment (sgouezel) [inline Mathlib/Analysis/SpecialFunctions/Trigonometric/Complex.lean:64]:
```suggestion
/-- The tangent of a complex number is equal to zero
```

File: Mathlib/Analysis/SpecialFunctions/Trigonometric/Complex.lean [hunk 3]
```diff
@@ -78,6 +80,13 @@ theorem tan_int_mul_pi_div_two (n : ℤ) : tan (n * π / 2) = 0 :=
   tan_eq_zero_iff.mpr (by use n)
 #align complex.tan_int_mul_pi_div_two Complex.tan_int_mul_pi_div_two
 
+/-- If tangent of a complex number is well-defined,
```
Comment (sgouezel) [inline Mathlib/Analysis/SpecialFunctions/Trigonometric/Complex.lean:83]:
```suggestion
/-- If the tangent of a complex number is well-defined,
```

File: Mathlib/Analysis/SpecialFunctions/Trigonometric/Complex.lean [hunk 4]
```diff
@@ -78,6 +80,13 @@ theorem tan_int_mul_pi_div_two (n : ℤ) : tan (n * π / 2) = 0 :=
...
+then it is equal to zero iff the number is equal to `k * π` for an integer `k`.
+
+See also `Complex.tan_eq_zero_iff` for a version that takes into account junk values of `θ`. -/
+theorem tan_eq_zero_iff' {θ : ℂ} (hθ : cos θ ≠ 0) : tan θ = 0 ↔ ∃ k : ℤ, θ = k * π := by
```
Comment (sgouezel) [inline Mathlib/Analysis/SpecialFunctions/Trigonometric/Complex.lean:87]: Could you also add the real version, for symmetry?

File: Mathlib/Analysis/SpecialFunctions/Trigonometric/Complex.lean (hunk 1 above)
Comment (urkud) [reply Mathlib/Analysis/SpecialFunctions/Trigonometric/Complex.lean]: Done (for tangent, not for other trig functions yet).

File: Mathlib/Analysis/SpecialFunctions/Trigonometric/Complex.lean (hunk 4 above)
Comment (urkud) [reply Mathlib/Analysis/SpecialFunctions/Trigonometric/Complex.lean]: Done.

**Comment by sgouezel** [comment]
bors r+
Thanks!

**Pull request closed without merging**